src/bindings/python/src/pyopenvino/core/remote_context.hpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include <pybind11/pybind11.h>
+
+#include <openvino/core/any.hpp>
+#include <openvino/core/type/element_type.hpp>
+#include <openvino/runtime/intel_gpu/properties.hpp>
+#include <openvino/runtime/intel_gpu/remote_properties.hpp>
+#include <openvino/runtime/remote_context.hpp>
+#include <openvino/runtime/tensor.hpp>
+
+#include "openvino/core/except.hpp"
+#include "pyopenvino/core/remote_tensor.hpp"
+
+#ifdef PY_ENABLE_GPU
+#ifndef _WIN32
+#    include <va/va.h>
+#endif  // _WIN32
+#endif  // PY_ENABLE_GPU
+
+namespace py = pybind11;
+
+class RemoteContextWrapper {
+public:
+    RemoteContextWrapper() {}
+
+    RemoteContextWrapper(ov::RemoteContext& _context): context{_context} {}
+
+    RemoteContextWrapper(ov::RemoteContext&& _context): context{std::move(_context)} {}
+
+    ov::RemoteContext context;
+};
+
+void regclass_RemoteContext(py::module m);
+
+#ifdef PY_ENABLE_GPU
+class ClContextWrapper : public RemoteContextWrapper {
+public:
+    ClContextWrapper(ov::RemoteContext& _context): RemoteContextWrapper{_context} {}
+
+    ClContextWrapper(ov::RemoteContext&& _context): RemoteContextWrapper{std::move(_context)} {}
+};
```

Comment:
looks like the code above (`ClContextWrapper`) does not depend on GPU. Why do we need condition for `PY_ENABLE_GPU` then?

Reply:
Removed this condition all together.